### ISSUE TITLE
SQL71501 Check Constraint has an unresolved reference to ARRAY

### ISSUE THREAD

**Issue by phil000** [microsoft/DacFx#519]
- SqlPackage or DacFx Version: SQL Server Data Tools   17.11.47.0
- .NET Framework (Windows-only) or .NET Core: .NET Core 8
- Environment (local platform and source/target platforms):

**Steps to Reproduce:**

1. Create table using ISJSON check constraint, using the newer 'json_type_constraint' parameter
2. You will get the error SQL71501: _Check Constraint: 'x' has an unresolved reference to object [dbo].[RegionProxy].[ARRAY]_
3. Using the ISJSON without the 'json_type_constraint' parameter does not cause the error

**Did this occur in prior versions?  If not - which version(s) did it work in?**
Not sure. Never needed ISJSON constraint prior to this.

(DacFx/SqlPackage/SSMS/Azure Data Studio)
Visual Studio 2022 17.11.5

e.g.
```
CREATE TABLE [dbo].[RegionProxy] (
    [RegionProxyId]        INT            IDENTITY (1, 1) NOT NULL,
    [RegionId]             INT            NOT NULL,
    [RegionName]           VARCHAR (150)  NOT NULL,
    [StandardWorkingDays]  NVARCHAR (100) NULL,
    [Version]              ROWVERSION     NOT NULL,
    CONSTRAINT [PK_RegionProxy_RegionProxyId] PRIMARY KEY CLUSTERED ([RegionProxyId] ASC),
    CONSTRAINT [CK_RegionProxy_StandardWorkingDays] CHECK (isjson([StandardWorkingDays],ARRAY)>(0))
);
```

**Comment by ErikEJ** [comment]
@phil000 Have you tried VS 17.12 preview?

**Comment by phil000** [comment]
No sorry I have not. I'm not really able to use a preview version.

**Comment by phil000** [comment]
@ErikEJ I've upgraded to VS 17.12.0 now it is released and the issue has been resolved.

This is working

```sql
CONSTRAINT [CK_RegionProxy_StandardWorkingDays] CHECK (isjson([StandardWorkingDays], ARRAY)>(0)),
```

**Comment by phil000** [comment]
However I still get the same thing on the Azure build agents. I presume we just need to wait for them to update the installed software.

##[error]Tables\RegionProxy.sql(11,90): Error SQL71501: Check Constraint: [dbo].[CK_RegionProxy_StandardWorkingDays] has an unresolved reference to object [dbo].[RegionProxy].[ARRAY]

**Comment by dzsquared** [comment]
Closing this out as the fix was part of DacFx/SqlPackage 162.4 (September 2024), which landed in Visual Studio 17.12
